test/functional/test_framework/script.py

```diff
@@ -731,3 +731,37 @@ def test_bn2vch(self):
         self.assertEqual(bn2vch(0xFFFFFFFF), bytes([0xFF, 0xFF, 0xFF, 0xFF, 0x00]))
         self.assertEqual(bn2vch(123456789), bytes([0x15, 0xCD, 0x5B, 0x07]))
         self.assertEqual(bn2vch(-54321), bytes([0x31, 0xD4, 0x80]))
+
+    def test_mining(self):
+        self.mine_chain()
+        node = self.nodes[0]
+
+        self.log.info('getmininginfo')
+        mining_info = node.getmininginfo()
+        self.assert_equal(mining_info['blocks'], 200)
+        self.assert_equal(mining_info['chain'], self.chain)
+        assert 'currentblocktx' not in mining_info
+        assert 'currentblockweight' not in mining_info
+        self.assert_equal(mining_info['difficulty'], Decimal('4.656542373906925E-10'))
+        self.assert_equal(mining_info['networkhashps'], Decimal('0.003333333333333334'))
+        self.assert_equal(mining_info['pooledtx'], 0)
+
+        # Mine a block to leave initial block download
+        node.generatetoaddress(1, node.get_deterministic_priv_key().address)
+        tmpl = node.getblocktemplate({'rules': ['segwit']})
+        self.log.info("getblocktemplate: Test capability advertised")
+        assert 'proposal' in tmpl['capabilities']
+        assert 'coinbasetxn' not in tmpl
+
+        next_height = int(tmpl["height"])
+        coinbase_tx = create_coinbase(height=next_height)
+        # sequence numbers must not be max for nLockTime to have effect
+        coinbase_tx.vin[0].nSequence = 2 ** 32 - 2
+        coinbase_tx.rehash()
+
+        # round-trip the encoded bip34 block height commitment
+        self.assert_equal(CScriptNum.decode(coinbase_tx.vin[0].scriptSig), next_height)
```

Comment:
I think for a unit test it could make sense to simply hard-code the value of `coinbase_tx.vin[0].scriptSig` instead of computing it every single time by mining 200 blocks.